### PR TITLE
Andres Chavez

### LINKED ISSUE #11: Add endpoints to retrieve and update user info
We need a back-end API endpoints to retrieve and update user account information such as:

- Home address
- Display name for book rating and commenting
- Credit card information (allow for multiple)
- Shipping address (allow for multiple)

### PR DESCRIPTION
**Pull request by AndresChavez97**
Pull Request to add User API to project

Fixes #11

### CODE REVIEW

File: api/src/AspNetCoreDemoApp/Program.cs [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 using Microsoft.AspNetCore.Hosting;
+using System;
 
 namespace AspNetCoreDemoApp
 {
     public class Program
-    {
-        public static void Main(string[] args) =>
+    { 
+        public static void Main(string[] args)
```
Comment (rconc016) [inline api/src/AspNetCoreDemoApp/Program.cs]: This change seems unnecessary.

File: api/src/AspNetCoreDemoApp/Services/IUserService.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using System.Threading;
+using AspNetCoreDemoApp.Models;
+using Google.Cloud.Firestore;
+
+namespace AspNetCoreDemoApp.Services
+{
+    public interface IUserService
+    {
+        void Create(string userId, User userData);
```
Comment (rconc016) [inline api/src/AspNetCoreDemoApp/Services/IUserService.cs]: You should add missing documentation for this method just for consistency.

File: api/src/AspNetCoreDemoApp/Services/IFirestoreService.cs [hunk 3]
```diff
@@ -13,10 +13,11 @@ public interface IFirestoreService
         /// Creates a new document in the Firestore database.
         /// </summary>
         /// <param name="collectionId">The ID of the parent collection.</param>
+        /// <param name="documentId">The id of the document being created</param>
         /// <param name="documentData">The data to be stored.</param>
         /// <param name="options">Optional set of options to be used when creating the document.</param>
         /// <param name="cancellationToken">Optional task cancellation token.</param>
-        void Create(string collectionId, object documentData, SetOptions options = null, CancellationToken cancellationToken = default(CancellationToken));
+        void Create(string collectionId, string documentId, object documentData, SetOptions options = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (rconc016) [inline api/src/AspNetCoreDemoApp/Services/IFirestoreService.cs]: You should overload the method to have two versions. One with the document ID and one without. This way we can still create documents with auto-generated IDs.

File: api/src/AspNetCoreDemoApp/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            this.userService = userService;
+        }
+
+        /// <summary>
+		/// GET endpoint to retrieve the list of all the users.
+		/// </summary>
+		/// <returns>A list of all <see href="User">s available.</returns>
+        [HttpGet]
+        public IEnumerable<User> Get()
```
[the remark this answers is not in the document]
Comment (AndresChavez97) [reply api/src/AspNetCoreDemoApp/Controllers/UserController.cs]: I was using it for testing purposes, I don't think we will need it in practice though.

[the remark this answers is not in the document]
Comment (rconc016) [reply api/src/AspNetCoreDemoApp/Controllers/UserController.cs]: We should probably remove if we're not going to need it.

[the remark this answers is not in the document]
Comment (AndresChavez97) [reply api/src/AspNetCoreDemoApp/Controllers/UserController.cs]: Going to leave endpoint for all users for now, we can always remove it down the line if we end up never using it.

**Pull request merged**